enhancements/builds/volume-mounted-resources.md

```diff
@@ -0,0 +1,297 @@
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. Can we make this the default or even only behavior for builds?
+   No, need to make it opt-in to avoid potentially breaking existing buildconfigs.
+2. What happens if a user overrides the default volume mounts used by Builds today?
+   We override the default with the sources chosen in the BuildConfig volumes array.
+
+## Summary
+
+Today builds support getting source input from configmaps and secrets, hereafter referred to as "resources".
+When users utilize this feature, the resource is volume-mounted into the build pod and placed in the "build context" within the build's execution environment, alongside other build sources like git source code.
+The next steps depend on whether it is an s2i or dockerfile build.
+
+For s2i builds, the generated Dockerfile contains commands to `ADD` the content at a path specified by the user, the assemble script is invoked, and then the injected content is zeroed out prior to committing the image via a `RUN rm` command added to the Dockerfile.
+
+For dockerfile builds, the user is instructed to add appropriate `ADD` and `RUN rm` commands to their dockerfile to inject the content that is available in the build's working directory (along with their application source, where applicable).
+
+There are a few undesirable aspects to this:
+
+1. In the dockerfile case, the content can still be found in lower layers of the image unless a layer squashing option is selected.
+2. Requires extra work by the user in the Dockerfile, so each Dockerfile must be customized
+
+This enhancement proposes to introduce an option to use buildah's capability to mount a volume at build time.
+The content mounted into the build pod would be then mounted into the container processing the Dockerfile, making that content available within the container so Dockerfile commands could reference it.
+No explicit `ADD` would be required, and since mounted content is not committed to the resulting image, no `RUN rm` equivalent is required to clean up the injected content.
+
+To avoid security and lifecycle concerns, the following volume types will be supported initially:
+
+1. Secrets
+2. ConfigMaps
+3. `csi` - this is to enable use of [Projected Resource CSI driver](/enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md) volumes.
+
+
+## Motivation
+
+### Goals
+
+* Simplify how users consume secret + configmap content in builds
+* Increase the security of protected content being injected to images
+* Simplify use cases that require consuming credentials during the build, but need to ensure those credentials do not end up in the output image.
+* Eventually extend this api to allow the mounting of traditional volumes (such as those backed by persistent storage)
```

Comment:
should some reference to @bparees 's original volume mount proposal that you noted in the PR description be made here?

I'm assuming that is at least part of the future being referenced here

I'm also assuming there is no superceding or replacement of @bparees 's original EPs in this space.  Just some "augmentation"

But some brief text describing the relationships, overlapping, non-overlapping elements, even some sort of roadmap-ish flow seems useful and possibly even necessary here

Reply:
This is a true co-authoriship - a big chunk of the text is Ben's original proposal. That never merged, I linked the two via GitHub comments.